### PR TITLE
Adding OpenCL implementation to CircusTent main branch

### PR DESCRIPTION
**Pull request by michael-beebe**
Pull request to merge the OpenCL additions to the master branch. Let me know if there are any issues or anything you'd like me to change/do differently.

Email: michael.beebe@ttu.edu

### CODE REVIEW

**Comment by jleidel** [comment]
`src/CircusTent/.DS_Store` should be removed from the directory

File: src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * \class CT_OPENCL
+ *
+ * \ingroup CircusTent
+ *
+ * \brief CircusTent OpenCL Implementation
+ *
+ */
+
+#ifdef _ENABLE_OPENCL_      // todo
```
Comment (jleidel) [inline src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.h]: remove the `todo`

File: src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+#ifndef _CT_OPENCL_H_       // todo
```
Comment (jleidel) [inline src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.h]: remove the `todo`

File: src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+#define _CT_OPENCL_H_       // todo
```
Comment (jleidel) [inline src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.h]: remove the `todo`

File: src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.cpp [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+      break;
+    }
+  }
+  else if (BType == CT_SG)
+  {
+    switch (AType)
+    {
+    case CT_ADD:
+      cl_kernel SG_ADD = clCreateKernel(program, "SG_ADD", NULL);
+       // XXX: arg might need to be a pointer
```
Comment (BrodyWilliams) [inline src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.cpp]: Probably best to remove this note.

File: src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.cpp [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+ * contact@tactcomplabs.com
+ *
+ * See LICENSE in the top level directory for licensing details
+ */
+
+#ifdef _CT_OPENCL_H_
+#include "CT_OPENCL.h"
+#include <fstream>
+
+#define __CL_ENABLE_EXCEPTIONS
```
Comment (BrodyWilliams) [inline src/CircusTent/Impl/CT_OPENCL/CT_OPENCL.cpp]: These includes are in CT_OPENCL.h already I think.

**Comment by jleidel** [comment]
I believe this resolves all my issues.  @BrodyWilliams, are we good to merge?

**Review by BrodyWilliams — APPROVED** [review]

**Comment by BrodyWilliams** [comment]
Looks correct on my end.

**Review by jleidel — APPROVED** [review]

**Pull request merged**